src/rdkafka_conf.c

```diff
@@ -3391,6 +3429,21 @@ const char *rd_kafka_topic_conf_finalize (rd_kafka_type_t cltype,
                 } else {
                         tconf->queuing_strategy = RD_KAFKA_QUEUE_FIFO;
                 }
+
+                if (conf->eos.transactional_id) {
+                        /* Make sure at least one message can be produced
+                         * before the transaction times out. */
+                        if (conf->eos.transaction_timeout_ms - 1000 <=
```

Comment:
Nit: conf->eos.transaction_timeout_ms  >=  tconf->message_timeout_ms + 1000 is easier to read IMO. 

> The transaction timeout must be at least 1000 ms larger than `message.timeout.ms` and `socket.timeout.ms